src/Widgets/MonitorLabel.vala

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2024 elementary, Inc. <https://elementary.io>
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+public class Gala.MonitorLabel : Clutter.Actor {
+    private const string PROVIDER_TEMPLATE = """
+    @define-color BG_COLOR %s;
+    """;
+    private const string CAPTION_FONT_NAME = "Inter";
+    private const int SPACING = 12;
+
+    public Meta.Display display { get; construct; }
+    public MonitorLabelInfo label_info { get; construct; }
+
+    private Gtk.WidgetPath widget_path;
+    private Gtk.StyleContext style_context;
+    private Clutter.Canvas canvas;
+    private float scaling_factor = 1.0f;
+
+    public MonitorLabel (Meta.Display display, MonitorLabelInfo label_info) {
+        Object (
+            display: display,
+            label_info: label_info
+        );
+    }
+
+    construct {
+        scaling_factor = display.get_monitor_scale (label_info.monitor);
+
+        canvas = new Clutter.Canvas ();
+        canvas.scale_factor = scaling_factor;
+        set_content (canvas);
+
+        create_components ();
+
+        unowned var monitor_manager = display.get_context ().get_backend ().get_monitor_manager ();
+        monitor_manager.monitors_changed.connect (() => {
+            var current_scale = display.get_monitor_scale (label_info.monitor);
+            if (current_scale != scaling_factor) {
+                scaling_factor = current_scale;
+                canvas.scale_factor = scaling_factor;
+                create_components ();
+            }
+        });
+
+        canvas.draw.connect (draw);
+    }
+
+    private bool draw (Cairo.Context ctx, int width, int height) {
+        if (style_context == null) { // gtk is not initialized yet
+            create_gtk_objects ();
```

Comment:
if we are gonna to use gtk here, isn't better to use the daemon to create the labels? we should be reducing our usage of Gtk from inside the gala process, not increasing it.